File: 205-isomorphic-strings/isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        
        # isomorphic means s should be replacesd with t and t should be replace with s
        
        hm1 = {}
        hm2 = {}

        for i in range(len(s)):
            if s[i] not in hm1:
                hm1[s[i]] = {t[i]}
            else:
                hm1[s[i]].add(t[i])
        for i in hm1.values():
            if len(i)>1:
                return False
        # print(hm1)
        hm1 = {}
        for i in s:
            if i not in hm1:
                hm1[i] = 0
            hm1[i]+=1
        for i in t:
            if i not in hm2:
                hm2[i] = 0
            hm2[i]+=1

        r1 = ''
        for i in s:
            r1+=str(hm1[i])
        r2 = ''
        for i in t:
            r2+=str(hm2[i])
        return r1==r2

File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character, Character> hm = new HashMap<Character, Character>();
        HashMap<Character, Character> hm1 = new HashMap<Character, Character>();
        if(s.length()!=t.length())
            return false;
        
        for(int i=0;i<s.length();i++){
            if(hm.containsKey(s.charAt(i))&&(hm.get(s.charAt(i)))!=t.charAt(i)){
                return false;
            }
            hm.put(s.charAt(i),t.charAt(i));
        }
        for(int i=0;i<t.length();i++){
            if(hm1.containsKey(t.charAt(i)) && hm1.get(t.charAt(i))!=s.charAt(i)){
                return false;
            }
            hm1.put(t.charAt(i),s.charAt(i));
        }
        String res="";
        for(int i=0;i<s.length();i++){
            res+=hm.get(s.charAt(i));
        }
        System.out.println(res+" "+t);
        if(res.equals(t))
            return true;
        return false;
    }

}